rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -136,6 +136,10 @@ public Integer call() throws IOException
       log.debug(String.format("Read %d bytes from cached file", read));
     }
     catch (Exception ex) {
+      if (ex instanceof CancelledException) {
```

Comment:
can we also avoid reading from `remoteFS` in the condition in L125 and throw an exception instead? That would also ensure that if a CachedRRC is cancelled while execution, then we do not read from remote.

Reply:
If CachedRRC is cancelled while execution then we will not hit L125. And for the case we hit L125, invalidation is the right thing to do given we read lesser amount than we knew at the start we had in cache.

Reply:
Ok, yes. Even if the request is cancelled after we've hit L125, we should invalidate. 
I guess we could just throw a non-cancelled exception post L125 for the `catch` to handle all the remote read and invalidation handling part - but can be skipped for future too. I think we can also just do invalidation incase of cancellation + inconsistent read and not read the data at all from remote.

Reply:
Agreed, we can short-circuit the readFromRemote to save read for one read request, doesn't save us a lot though.

Reply:
This is now handled